### ISSUE TITLE
Intermittent crashing when building Xamarin Forms Control Gallery

### ISSUE THREAD

**Issue by PureWeen** [mono/monodevelop#5184]
I was testing some changes to the control galleries project file by deleting out this custom command

https://github.com/xamarin/Xamarin.Forms/blob/master/Xamarin.Forms.ControlGallery.Android/Xamarin.Forms.ControlGallery.Android.csproj#L59

And every few builds VS for Mac would crash.

My steps were basically
-edit project file
-close vs for Mac
-delete bin/obj folders in the https://github.com/xamarin/Xamarin.Forms/blob/master/Xamarin.Forms.ControlGallery.Android
-open solution back up
-build Xamarin.Forms.ControlGallery.Android.csproj
-The build takes a bit and sometimes it would just cause VS to crash

I've attached the log from the crash.
Version 7.6 Preview (7.6 build 1773)

[Ide.2018-06-22__13-56-14.log.zip](https://github.com/mono/monodevelop/files/2133915/Ide.2018-06-22__13-56-14.log.zip)

> VS bug [#644694](https://devdiv.visualstudio.com/DevDiv/_workitems/edit/644694)

**Comment by mrward** [comment]
Does this still happen if the IoT addin is disabled or uninstalled?

Most of the IDE errors before the crash seem to be related to having Xamarin.IoT 7.5 installed.

```
MSBuild property evaluation failed: [Microsoft.Build.Utilities.ToolLocationHelper]::GetPathToStandardLibraries($(TargetFrameworkIdentifier), $(TargetFrameworkVersion), $(TargetFrameworkProfile), $(PlatformTarget), $(TargetFrameworkRootPath))
System.Reflection.TargetInvocationException: Exception has been thrown by the target of an invocation. ---> System.Reflection.TargetInvocationException: Exception has been thrown by the target of an invocation. ---> System.Reflection.TargetInvocationException: Exception has been thrown by the target of an invocation. ---> System.TypeInitializationException: The type initializer for 'Microsoft.Build.Collections.MSBuildNameIgnoreCaseComparer' threw an exception. ---> System.EntryPointNotFoundException: GetSystemInfo
  at (wrapper managed-to-native) 
```

Crash:

```
Stacktrace:

  at <unknown> <0xffffffff>
  at (wrapper managed-to-native) System.Runtime.InteropServices.GCHandle.CheckCurrentDomain (int) [0x00007] in <30f82f22c5df4a87974ae23bf2702f8f>:0
  at System.Runtime.InteropServices.GCHandle.op_Explicit (intptr) [0x00018] in /Users/builder/jenkins/workspace/build-package-osx-mono/2018-02/external/bockbuild/builds/mono-x64/mcs/class/corlib/System.Runtime.InteropServices/GCHandle.cs:134
  at System.Runtime.InteropServices.GCHandle.FromIntPtr (intptr) [0x00000] in /Users/builder/jenkins/workspace/build-package-osx-mono/2018-02/external/bockbuild/builds/mono-x64/mcs/class/corlib/System.Runtime.InteropServices/GCHandle.cs:176
  at System.IO.Compression.DeflateStreamNative.UnmanagedWrite (intptr,int,intptr) [0x00000] in /Users/builder/jenkins/workspace/build-package-osx-mono/2018-02/external/bockbuild/builds/mono-x64/mcs/class/System/System.IO.Compression/DeflateStream.cs:437
  at (wrapper native-to-managed) System.IO.Compression.DeflateStreamNative.UnmanagedWrite (intptr,int,intptr) [0x00023] in <81ba78c8dc794b7f9f7b530c53db0f84>:0
  at <unknown> <0xffffffff>
  at (wrapper managed-to-native) System.IO.Compression.DeflateStreamNative.CloseZStream (intptr) [0x00009] in <81ba78c8dc794b7f9f7b530c53db0f84>:0
  at System.IO.Compression.DeflateStreamNative/SafeDeflateStreamHandle.ReleaseHandle () [0x00000] in /Users/builder/jenkins/workspace/build-package-osx-mono/2018-02/external/bockbuild/builds/mono-x64/mcs/class/System/System.IO.Compression/DeflateStream.cs:563
  at System.Runtime.InteropServices.SafeHandle.DangerousReleaseInternal (bool) [0x000a3] in /Users/builder/jenkins/workspace/build-package-osx-mono/2018-02/external/bockbuild/builds/mono-x64/mcs/class/corlib/System.Runtime.InteropServices/SafeHandle.cs:227
  at System.Runtime.InteropServices.SafeHandle.InternalFinalize () [0x00008] in /Users/builder/jenkins/workspace/build-package-osx-mono/2018-02/external/bockbuild/builds/mono-x64/mcs/class/corlib/System.Runtime.InteropServices/SafeHandle.cs:163
  at System.Runtime.InteropServices.SafeHandle.Dispose (bool) [0x0000a] in /Users/builder/jenkins/workspace/build-package-osx-mono/2018-02/external/bockbuild/builds/mono-x64/mcs/class/referencesource/mscorlib/system/runtime/interopservices/safehandle.cs:262
  at System.Runtime.InteropServices.SafeHandle.Finalize () [0x00000] in /Users/builder/jenkins/workspace/build-package-osx-mono/2018-02/external/bockbuild/builds/mono-x64/mcs/class/referencesource/mscorlib/system/runtime/interopservices/safehandle.cs:199
  at (wrapper runtime-invoke) object.runtime_invoke_virtual_void__this__ (object,intptr,intptr,intptr) [0x0001f] in <30f82f22c5df4a87974ae23bf2702f8f>:0
```

**Comment by PureWeen** [comment]
Alright I got another crash with IoT disabled

[Ide.2018-06-25__13-34-42.log.zip](https://github.com/mono/monodevelop/files/2135521/Ide.2018-06-25__13-34-42.log.zip)

I still don't have the most consistent way to repro... 
I just keep rebuilding the Android Control Gallery project, close vs, delete bin obj, rebuild, rebuild, and then during one of the rebuilds it'll crash

I'll see if I can find a better path to reproduce

**Comment by mrward** [comment]
The crash in the other IDE log looks the same to me. It does not seem to be related to the Xamarin.IoT addin.

```
Stacktrace:

  at <unknown> <0xffffffff>
  at (wrapper managed-to-native) System.Runtime.InteropServices.GCHandle.CheckCurrentDomain (int) [0x00007] in <30f82f22c5df4a87974ae23bf2702f8f>:0
  at System.Runtime.InteropServices.GCHandle.op_Explicit (intptr) [0x00018] in /Users/builder/jenkins/workspace/build-package-osx-mono/2018-02/external/bockbuild/builds/mono-x64/mcs/class/corlib/System.Runtime.InteropServices/GCHandle.cs:134
  at System.Runtime.InteropServices.GCHandle.FromIntPtr (intptr) [0x00000] in /Users/builder/jenkins/workspace/build-package-osx-mono/2018-02/external/bockbuild/builds/mono-x64/mcs/class/corlib/System.Runtime.InteropServices/GCHandle.cs:176
  at System.IO.Compression.DeflateStreamNative.UnmanagedWrite (intptr,int,intptr) [0x00000] in /Users/builder/jenkins/workspace/build-package-osx-mono/2018-02/external/bockbuild/builds/mono-x64/mcs/class/System/System.IO.Compression/DeflateStream.cs:437
  at (wrapper native-to-managed) System.IO.Compression.DeflateStreamNative.UnmanagedWrite (intptr,int,intptr) [0x00023] in <81ba78c8dc794b7f9f7b530c53db0f84>:0
  at <unknown> <0xffffffff>
  at (wrapper managed-to-native) System.IO.Compression.DeflateStreamNative.CloseZStream (intptr) [0x00009] in <81ba78c8dc794b7f9f7b530c53db0f84>:0
  at System.IO.Compression.DeflateStreamNative/SafeDeflateStreamHandle.ReleaseHandle () [0x00000] in /Users/builder/jenkins/workspace/build-package-osx-mono/2018-02/external/bockbuild/builds/mono-x64/mcs/class/System/System.IO.Compression/DeflateStream.cs:563
  at System.Runtime.InteropServices.SafeHandle.DangerousReleaseInternal (bool) [0x000a3] in /Users/builder/jenkins/workspace/build-package-osx-mono/2018-02/external/bockbuild/builds/mono-x64/mcs/class/corlib/System.Runtime.InteropServices/SafeHandle.cs:227
  at System.Runtime.InteropServices.SafeHandle.InternalFinalize () [0x00008] in /Users/builder/jenkins/workspace/build-package-osx-mono/2018-02/external/bockbuild/builds/mono-x64/mcs/class/corlib/System.Runtime.InteropServices/SafeHandle.cs:163
  at System.Runtime.InteropServices.SafeHandle.Dispose (bool) [0x0000a] in /Users/builder/jenkins/workspace/build-package-osx-mono/2018-02/external/bockbuild/builds/mono-x64/mcs/class/referencesource/mscorlib/system/runtime/interopservices/safehandle.cs:262
  at System.Runtime.InteropServices.SafeHandle.Finalize () [0x00000] in /Users/builder/jenkins/workspace/build-package-osx-mono/2018-02/external/bockbuild/builds/mono-x64/mcs/class/referencesource/mscorlib/system/runtime/interopservices/safehandle.cs:199
  at (wrapper runtime-invoke) object.runtime_invoke_virtual_void__this__ (object,intptr,intptr,intptr) [0x0001f] in <30f82f22c5df4a87974ae23bf2702f8f>:0
```

**Comment by Therzok** [comment]
That's the builder crashing with a mono crash: https://github.com/mono/monodevelop/pull/5060/files

**Comment by mrward** [comment]
Need to check on the VS Mac side to see if the builder crash can be handled. Looking at the VS Mac version where the builder crash fix went in on for VS Mac it seems that it did not prevent VS Mac from terminating. @PureWeen By crash I am assuming you mean that VS Mac terminates.

IDE log with crash: 7.6 build 1773
Commit: https://github.com/mono/monodevelop/commit/f350c29e1ece3a03ae8c26f5954043a8d57201b1
Tagged: monodevelop-7.6.0.1760

**Comment by mrward** [comment]
Unable to build the Xamarin.Forms.ControlGallery.Android.csproj project from the 2.5.1 branch. With VS Mac 7.6.0.1977 and Mono 5.12.0.273 (or 5.10.1) the build fails with 'Custom command execution failed'. Looking at the MSBuild output I can see the following:

```
Executing: xbuild /t:SignAndroidPackage Xamarin.Forms.ControlGallery.Android/Xamarin.Forms.ControlGallery.Android.csproj

....

	Target _ResolveAssemblies:
: error : Could not load file or assembly 'NuGet.Common, Version=4.6.0.0, Culture=neutral, PublicKeyToken=31bf3856ad364e35' or one of its dependencies.
  at Xamarin.Android.Tasks.ResolveAssemblies.<Execute>b__44_0 () [0x0000e] in <96027da06dd5411688a31011072e256f>:0 
  at System.Threading.Tasks.Task.InnerInvoke () [0x0000f] in <bb7b695b8c6246b3ac1646577aea7650>:0 
  at System.Threading.Tasks.Task.Execute () [0x00010] in <bb7b695b8c6246b3ac1646577aea7650>:0 
	Task "ResolveAssemblies" execution -- FAILED
	Done building target "_ResolveAssemblies" in project "Xamarin.Forms.ControlGallery.Android/Xamarin.Forms.ControlGallery.Android.csproj".-- FAILED
Done building project "Xamarin.Forms.ControlGallery.Android/Xamarin.Forms.ControlGallery.Android.csproj".-- FAILED

Build FAILED.
```

I can see the Mono crash being logged in the IDE log. VS Mac does not itself crash.

```
Stacktrace:

  at <unknown> <0xffffffff>
  at (wrapper managed-to-native) System.Runtime.InteropServices.GCHandle.CheckCurrentDomain (int) [0x00007] in <30f82f22c5df4a87974ae23bf2702f8f>:0
  at System.Runtime.InteropServices.GCHandle.op_Explicit (intptr) [0x00018] in /Users/builder/jenkins/workspace/build-package-osx-mono/2018-02/external/bockbuild/builds/mono-x64/mcs/class/corlib/System.Runtime.InteropServices/GCHandle.cs:134
  at System.Runtime.InteropServices.GCHandle.FromIntPtr (intptr) [0x00000] in /Users/builder/jenkins/workspace/build-package-osx-mono/2018-02/external/bockbuild/builds/mono-x64/mcs/class/corlib/System.Runtime.InteropServices/GCHandle.cs:176
  at System.IO.Compression.DeflateStreamNative.UnmanagedWrite (intptr,int,intptr) [0x00000] in /Users/builder/jenkins/workspace/build-package-osx-mono/2018-02/external/bockbuild/builds/mono-x64/mcs/class/System/System.IO.Compression/DeflateStream.cs:437
  at (wrapper native-to-managed) System.IO.Compression.DeflateStreamNative.UnmanagedWrite (intptr,int,intptr) [0x00023] in <81ba78c8dc794b7f9f7b530c53db0f84>:0
  at <unknown> <0xffffffff>
  at (wrapper managed-to-native) System.IO.Compression.DeflateStreamNative.CloseZStream (intptr) [0x00009] in <81ba78c8dc794b7f9f7b530c53db0f84>:0
  at System.IO.Compression.DeflateStreamNative/SafeDeflateStreamHandle.ReleaseHandle () [0x00000] in /Users/builder/jenkins/workspace/build-package-osx-mono/2018-02/external/bockbuild/builds/mono-x64/mcs/class/System/System.IO.Compression/DeflateStream.cs:563
  at System.Runtime.InteropServices.SafeHandle.DangerousReleaseInternal (bool) [0x000a3] in /Users/builder/jenkins/workspace/build-package-osx-mono/2018-02/external/bockbuild/builds/mono-x64/mcs/class/corlib/System.Runtime.InteropServices/SafeHandle.cs:227
  at System.Runtime.InteropServices.SafeHandle.InternalFinalize () [0x00008] in /Users/builder/jenkins/workspace/build-package-osx-mono/2018-02/external/bockbuild/builds/mono-x64/mcs/class/corlib/System.Runtime.InteropServices/SafeHandle.cs:163
  at System.Runtime.InteropServices.SafeHandle.Dispose (bool) [0x0000a] in /Users/builder/jenkins/workspace/build-package-osx-mono/2018-02/external/bockbuild/builds/mono-x64/mcs/class/referencesource/mscorlib/system/runtime/interopservices/safehandle.cs:262
  at System.Runtime.InteropServices.SafeHandle.Finalize () [0x00000] in /Users/builder/jenkins/workspace/build-package-osx-mono/2018-02/external/bockbuild/builds/mono-x64/mcs/class/referencesource/mscorlib/system/runtime/interopservices/safehandle.cs:199
  at (wrapper runtime-invoke) object.runtime_invoke_virtual_void__this__ (object,intptr,intptr,intptr) [0x0001f] in <30f82f22c5df4a87974ae23bf2702f8f>:0

Native stacktrace:

	0   mono64                              0x0000000106ed9b78 mono_handle_native_crash + 264
	1   mono64                              0x0000000106f45936 altstack_handle_and_restore + 70
	2   mono64                              0x00000001070a2516 mono_gchandle_is_in_domain + 22
	3   mono64                              0x0000000107090b17 mono_gc_GCHandle_CheckCurrentDomain + 23
	4   ???                                 0x000000010c5e68d6 0x0 + 4502481110
	5   libMonoPosixHelper.dylib            0x000000010c5a6ea0 CloseZStream + 128
	6   ???                                 0x000000010c5e66ed 0x0 + 4502480621
	7   mscorlib.dll.dylib                  0x00000001093a5ca1 System_Runtime_InteropServices_SafeHandle_InternalFinalize + 33
	8   ???                                 0x000000010b4597ac 0x0 + 4484077484
	9   mono64                              0x000000010708fd17 mono_gc_run_finalize + 727
	10  mono64                              0x00000001070ae5cc sgen_gc_invoke_finalizers + 236
	11  mono64                              0x0000000107091728 finalizer_thread + 728
	12  mono64                              0x000000010703f203 start_wrapper + 707
	13  libsystem_pthread.dylib             0x00007fffe877c93b _pthread_body + 180
	14  libsystem_pthread.dylib             0x00007fffe877c887 _pthread_body + 0
	15  libsystem_pthread.dylib             0x00007fffe877c08d thread_start + 13

Debug info from gdb:

(lldb) command source -s 0 '/tmp/mono-gdb-commands.oISv35'
Executing commands in '/tmp/mono-gdb-commands.oISv35'.
(lldb) process attach --pid 1754
warning: (x86_64) /Library/Frameworks/Mono.framework/Versions/5.12.0/lib/mono/4.5/mscorlib.dll.dylib empty dSYM file detected, dSYM was created with an executable with no debug info.
Process 1754 stopped
* thread #1, name = 'tid_307', queue = 'com.apple.main-thread', stop reason = signal SIGSTOP
    frame #0: 0x00007fffe868a39e libsystem_kernel.dylib`semaphore_timedwait_trap + 10
libsystem_kernel.dylib`semaphore_timedwait_trap:
->  0x7fffe868a39e <+10>: retq   
    0x7fffe868a39f <+11>: nop    

libsystem_kernel.dylib`semaphore_timedwait_signal_trap:
    0x7fffe868a3a0 <+0>:  movq   %rcx, %r10
    0x7fffe868a3a3 <+3>:  movl   $0x1000027, %eax          ; imm = 0x1000027 
Target 0: (mono64) stopped.

Executable module set to "/Library/Frameworks/Mono.framework/Versions/5.12.0/bin/mono64".
Architecture set to: x86_64h-apple-macosx.
(lldb) thread list
Process 1754 stopped
* thread #1: tid = 0xf950, 0x00007fffe868a39e libsystem_kernel.dylib`semaphore_timedwait_trap + 10, name = 'tid_307', queue = 'com.apple.main-thread', stop reason = signal SIGSTOP
  thread #2: tid = 0xf952, 0x00007fffe8691bf2 libsystem_kernel.dylib`__psynch_cvwait + 10, name = 'SGen worker'
  thread #3: tid = 0xf953, 0x00007fffe86923ee libsystem_kernel.dylib`__wait4 + 10, name = 'Finalizer'
  thread #4: tid = 0xf954, 0x00007fffe869244e libsystem_kernel.dylib`__workq_kernreturn + 10
(lldb) thread backtrace all
Process 1754 exited with status = 0 (0x00000000) Terminated due to signal 9
warning: could not execute support code to read Objective-C class data in the process. This may reduce the quality of type information available.
* thread #1, name = 'tid_307', queue = 'com.apple.main-thread', stop reason = signal SIGSTOP
  * frame #0: 0x00007fffe868a39e libsystem_kernel.dylib`semaphore_timedwait_trap + 10
    frame #1: 0x0000000107090510 mono64`mono_coop_sem_timedwait [inlined] mono_os_sem_timedwait(timeout_ms=<unavailable>) at mono-os-semaphore.h:124 [opt]
    frame #2: 0x00000001070904fc mono64`mono_coop_sem_timedwait(sem=0x00007ff816c84fb0, timeout_ms=<unavailable>, flags=<unavailable>) at mono-coop-semaphore.h:57 [opt]
    frame #3: 0x0000000107090067 mono64`mono_domain_finalize(domain=<unavailable>, timeout=4294967295) at gc.c:467 [opt]
    frame #4: 0x0000000106e2d822 mono64`mini_cleanup(domain=0x00007ff814506270) at mini-runtime.c:4457 [opt]
    frame #5: 0x0000000106e9de8e mono64`mono_main(argc=<unavailable>, argv=<unavailable>) at driver.g.c:2431 [opt]
    frame #6: 0x0000000106e1be58 mono64`main [inlined] mono_main_with_options(argc=<unavailable>, argv=<unavailable>) at main.c:50 [opt]
    frame #7: 0x0000000106e1be44 mono64`main(argc=5, argv=<unavailable>) at main.c:398 [opt]
    frame #8: 0x0000000106e1bd44 mono64`start + 52
  thread #2, name = 'SGen worker'
    frame #0: 0x00007fffe8691bf2 libsystem_kernel.dylib`__psynch_cvwait + 10
    frame #1: 0x00007fffe877d7fa libsystem_pthread.dylib`_pthread_cond_wait + 712
    frame #2: 0x00000001070e397e mono64`thread_func [inlined] mono_os_cond_wait(mutex=<unavailable>) at mono-os-mutex.h:173 [opt]
    frame #3: 0x00000001070e396b mono64`thread_func at sgen-thread-pool.c:165 [opt]
    frame #4: 0x00000001070e395d mono64`thread_func(data=0x0000000000000000) at sgen-thread-pool.c:196 [opt]
    frame #5: 0x00007fffe877c93b libsystem_pthread.dylib`_pthread_body + 180
    frame #6: 0x00007fffe877c887 libsystem_pthread.dylib`_pthread_start + 286
    frame #7: 0x00007fffe877c08d libsystem_pthread.dylib`thread_start + 13
  thread #3, name = 'Finalizer'
    frame #0: 0x00007fffe86923ee libsystem_kernel.dylib`__wait4 + 10
    frame #1: 0x0000000106ed9d58 mono64`mono_handle_native_crash(signal=<unavailable>, ctx=<unavailable>, info=<unavailable>) at mini-exceptions.c:3023 [opt]
    frame #2: 0x0000000106f45936 mono64`altstack_handle_and_restore(ctx=0x000070000bc80700, obj=0x0000000000000000, stack_ovf=0) at exceptions-amd64.c:857 [opt]
    frame #3: 0x00000001070a2516 mono64`mono_gchandle_is_in_domain(gchandle=<unavailable>, domain=0x00007ff814506270) at sgen-mono.c:2532 [opt]
    frame #4: 0x0000000107090b17 mono64`mono_gc_GCHandle_CheckCurrentDomain(gchandle=<unavailable>) at gc.c:695 [opt]
    frame #5: 0x000000010c5e68d6
    frame #6: 0x000000010c5a6ea0 libMonoPosixHelper.dylib`CloseZStream [inlined] write_to_managed(stream=0x00007ff8147f02d0) at zlib-helper.c:147 [opt]
    frame #7: 0x000000010c5a6e7c libMonoPosixHelper.dylib`CloseZStream [inlined] flush_internal(stream=0x00007ff8147f02d0, is_final=1) at zlib-helper.c:169 [opt]
    frame #8: 0x000000010c5a6e74 libMonoPosixHelper.dylib`CloseZStream(zstream=0x00007ff8147f02d0) at zlib-helper.c:123 [opt]
    frame #9: 0x000000010c5e66ed
    frame #10: 0x00000001093a5ca1 mscorlib.dll.dylib`System_Runtime_InteropServices_SafeHandle_InternalFinalize + 33
    frame #11: 0x00000001093a5a6a mscorlib.dll.dylib`System_Runtime_InteropServices_SafeHandle_Dispose_bool + 42
    frame #12: 0x00000001093a594b mscorlib.dll.dylib`System_Runtime_InteropServices_SafeHandle_Finalize + 27
    frame #13: 0x000000010b4597ac
    frame #14: 0x000000010708fd17 mono64`mono_gc_run_finalize(obj=<unavailable>, data=<unavailable>) at gc.c:318 [opt]
    frame #15: 0x00000001070ae5cc mono64`sgen_gc_invoke_finalizers at sgen-gc.c:2765 [opt]
    frame #16: 0x0000000107091728 mono64`finalizer_thread [inlined] finalize_domain_objects at gc.c:822 [opt]
    frame #17: 0x00000001070916f3 mono64`finalizer_thread [inlined] mono_runtime_do_background_work at gc.c:849 [opt]
    frame #18: 0x000000010709169e mono64`finalizer_thread(unused=<unavailable>) at gc.c:899 [opt]
    frame #19: 0x000000010703f203 mono64`start_wrapper [inlined] start_wrapper_internal at threads.c:1073 [opt]
    frame #20: 0x000000010703f173 mono64`start_wrapper(data=0x00007ff8146010e0) at threads.c:1133 [opt]
  thread #4
    frame #0: 0xffffffffffffffff __lldb_caller_function`__lldb_caller_function[
(lldb) detach
error: Process must be launched.
```

**Comment by mrward** [comment]
The custom tools build failure seems like a red herring. If you comment out that part the build succeeds with one warning but the IDE log still has the mono crash. The crash also happens in the MSBuild host that is not building the ControlGallery.Android.csproj.

Right clicking the Xamarin.Forms.ControlGallery.Android.csproj and selecting Build causes two MSBuild hosts to run builds for the projects.

```
Process 12023 (process lldb attaches to with Mono crash)

2018-07-09__14-23-21 BeginBuildOperation start - Xamarin.Forms/Xamarin.Forms.Core/Xamarin.Forms.Core.csproj
2018-07-09__14-23-21 BeginBuildOperation end - Xamarin.Forms/Xamarin.Forms.Core/Xamarin.Forms.Core.csproj
2018-07-09__14-23-21 ProjectBuilder.Run - start - TaskId 84, Targets: Build, File: Xamarin.Forms/Xamarin.Forms.Platform.Android.FormsViewGroup/Xamarin.Forms.Platform.Android.FormsViewGroup.csproj
2018-07-09__14-23-27 ProjectBuilder.Run - end - TaskId 84, Targets: Build, File: Xamarin.Forms/Xamarin.Forms.Platform.Android.FormsViewGroup/Xamarin.Forms.Platform.Android.FormsViewGroup.csproj
2018-07-09__14-23-27 ProjectBuilder.Run - start - TaskId 93, Targets: Build, File: Xamarin.Forms/Xamarin.Forms.Xaml/Xamarin.Forms.Xaml.csproj
2018-07-09__14-23-28 ProjectBuilder.Run - end - TaskId 93, Targets: Build, File: Xamarin.Forms/Xamarin.Forms.Xaml/Xamarin.Forms.Xaml.csproj
2018-07-09__14-23-28 ProjectBuilder.Run - start - TaskId 94, Targets: Build, File: Xamarin.Forms/Xamarin.Forms.Maps/Xamarin.Forms.Maps.csproj
2018-07-09__14-23-28 ProjectBuilder.Run - end - TaskId 94, Targets: Build, File: Xamarin.Forms/Xamarin.Forms.Maps/Xamarin.Forms.Maps.csproj
2018-07-09__14-23-28 ProjectBuilder.Run - start - TaskId 95, Targets: Build, File: Xamarin.Forms/Xamarin.Forms.Pages/Xamarin.Forms.Pages.csproj
2018-07-09__14-23-29 ProjectBuilder.Run - end - TaskId 95, Targets: Build, File: Xamarin.Forms/Xamarin.Forms.Pages/Xamarin.Forms.Pages.csproj
2018-07-09__14-23-29 ProjectBuilder.Run - start - TaskId 96, Targets: Build, File: Xamarin.Forms/Xamarin.Forms.Platform.Android.AppLinks/Xamarin.Forms.Platform.Android.AppLinks.csproj
2018-07-09__14-23-30 ProjectBuilder.Run - end - TaskId 96, Targets: Build, File: Xamarin.Forms/Xamarin.Forms.Platform.Android.AppLinks/Xamarin.Forms.Platform.Android.AppLinks.csproj
2018-07-09__14-23-30 ProjectBuilder.Run - start - TaskId 97, Targets: Build, File: Xamarin.Forms/Xamarin.Forms.Build.Tasks/Xamarin.Forms.Build.Tasks.csproj
2018-07-09__14-23-33 ProjectBuilder.Run - end - TaskId 97, Targets: Build, File: Xamarin.Forms/Xamarin.Forms.Build.Tasks/Xamarin.Forms.Build.Tasks.csproj
2018-07-09__14-23-33 ProjectBuilder.Run - start - TaskId 98, Targets: Build, File: Xamarin.Forms/Xamarin.Forms.Controls/Xamarin.Forms.Controls.csproj
2018-07-09__14-23-37 ProjectBuilder.Run - end - TaskId 98, Targets: Build, File: Xamarin.Forms/Xamarin.Forms.Controls/Xamarin.Forms.Controls.csproj
2018-07-09__14-24-04 ProjectBuilder.Run - start - TaskId 100, Targets: Build, File: Xamarin.Forms/Xamarin.Forms.Maps.Android/Xamarin.Forms.Maps.Android.csproj
2018-07-09__14-24-16 ProjectBuilder.Run - end - TaskId 100, Targets: Build, File: Xamarin.Forms/Xamarin.Forms.Maps.Android/Xamarin.Forms.Maps.Android.csproj

Process 11998

2018-07-09__14-23-20 BeginBuildOperation start - Xamarin.Forms/Xamarin.Forms.Core/Xamarin.Forms.Core.csproj
2018-07-09__14-23-20 BeginBuildOperation end - Xamarin.Forms/Xamarin.Forms.Core/Xamarin.Forms.Core.csproj
2018-07-09__14-23-20 ProjectBuilder.Run - start - TaskId 83, Targets: Build, File: Xamarin.Forms/Xamarin.Forms.Platform/Xamarin.Forms.Platform.csproj
2018-07-09__14-23-26 ProjectBuilder.Run - end - TaskId 83, Targets: Build, File: Xamarin.Forms/Xamarin.Forms.Platform/Xamarin.Forms.Platform.csproj
2018-07-09__14-23-26 ProjectBuilder.Run - start - TaskId 90, Targets: Build, File: Xamarin.Forms/Xamarin.Forms.CustomAttributes/Xamarin.Forms.CustomAttributes.csproj
2018-07-09__14-23-26 ProjectBuilder.Run - end - TaskId 90, Targets: Build, File: Xamarin.Forms/Xamarin.Forms.CustomAttributes/Xamarin.Forms.CustomAttributes.csproj
2018-07-09__14-23-26 ProjectBuilder.Run - start - TaskId 91, Targets: Build, File: Xamarin.Forms/Xamarin.Forms.Core/Xamarin.Forms.Core.csproj
2018-07-09__14-23-27 ProjectBuilder.Run - end - TaskId 91, Targets: Build, File: Xamarin.Forms/Xamarin.Forms.Core/Xamarin.Forms.Core.csproj
2018-07-09__14-23-27 ProjectBuilder.Run - start - TaskId 92, Targets: Build, File: Xamarin.Forms/Xamarin.Forms.Platform.Android/Xamarin.Forms.Platform.Android.csproj
2018-07-09__14-24-04 ProjectBuilder.Run - end - TaskId 92, Targets: Build, File: Xamarin.Forms/Xamarin.Forms.Platform.Android/Xamarin.Forms.Platform.Android.csproj
2018-07-09__14-24-04 ProjectBuilder.Run - start - TaskId 99, Targets: Build, File: Xamarin.Forms/Stubs/Xamarin.Forms.Platform.Android/Xamarin.Forms.Platform.Android (Forwarders).csproj
2018-07-09__14-24-22 ProjectBuilder.Run - end - TaskId 99, Targets: Build, File: Xamarin.Forms/Stubs/Xamarin.Forms.Platform.Android/Xamarin.Forms.Platform.Android (Forwarders).csproj
2018-07-09__14-24-22 ProjectBuilder.Run - start - TaskId 101, Targets: Build, File: Xamarin.Forms/Xamarin.Forms.ControlGallery.Android/Xamarin.Forms.ControlGallery.Android.csproj
2018-07-09__14-25-29 ProjectBuilder.Run - end - TaskId 101, Targets: Build, File: Xamarin.Forms/Xamarin.Forms.ControlGallery.Android/Xamarin.Forms.ControlGallery.Android.csproj
2018-07-09__14-25-29 EndBuildOperation start
2018-07-09__14-25-29 EndBuildOperation end

```

**Comment by PureWeen** [comment]
@mrward  yea in our main branch we've commented out that custom command but it's still there on 2.5.1

I wasn't able to get to the point where I was getting that exception though. I was trying to test something on 2.5.1 and whenever I right clicked to build that project VSfM would crash (so just completely close without any warning or error messages)

I was hoping you'd have a similar experience when trying to build 2.5.1 but it looks like your build was able to get a bit further

**Comment by xamarin-release-manager** [comment]
> [sync] [VS-26] Comment by **xamarinc**



Fixed in version 8.6.0.802 (master)



Author: therzok  
 Commit: 10ef2dab1ab2f317b4a054799c2b2759432a162d (xamarin/monodevelop)
